qiskit/pulse/model/pulse_target.py

```diff
@@ -20,15 +20,80 @@
 from qiskit.pulse.exceptions import PulseError
 
 
-class LogicalElement(ABC):
+class PulseTarget(ABC):
+    """Base class of pulse target.
+
+    When playing a pulse on a quantum hardware, one typically has to define on what hardware component
+    the pulse will be played, and the frame (frequency and phase) of the carrier wave.
+    :class:`PulseTarget` addresses the first of two, and identifies the component which is the target
+    of the pulse. Every played pulse and most other instructions are associated with a
+    :class:`PulseTarget` on which they are performed.
+    """
+
+    def __init__(self, hash_identifier):
+        """Create ``PulseTarget``.
+
+        Args:
+            hash_identifier: A hashable unique identifier.
+        """
+        self._hash = hash((hash_identifier, type(self)))
+
+    def __hash__(self) -> int:
+        return self._hash
+
+
+class Port(PulseTarget):
+    """A ``Port`` type ``PulseTarget``.
+
+    A :class:`Port` is the most basic ``PulseTarget`` - simply a hardware port responsible for
+    pulse emission. A :class:`Port` is identified by a string, which must be recognized by the
+    backend. Therefore, using pulse level control with :class:`Port` requires an extensive
+    knowledge of the hardware. When possible, it is recommended to use :class:`LogicalElement`,
```

Comment:
```suggestion
    knowledge of the hardware. When possible, it is recommended to use :class:`LogicalElement`,
```
I don't think we need to recommend some use pattern in the class documentation (this is more like tutorial). We just need to tell the identifier cannot be determined by the end users, and user-provided identifier will just fail in execution. e.g. if you are a pulse compiler pass developer, dealing with `Port` type might be more convenient.

Reply:
Thanks.